edx_sga/sga.py

```diff
@@ -434,14 +430,67 @@ def staff_download_annotated(self, request, suffix=''):
         )
 
     def download(self, path, mimetype, filename):
-        BLOCK_SIZE = (1 << 10) * 8  # 8kb
-        file = default_storage.open(path)
-        app_iter = iter(partial(file.read, BLOCK_SIZE), '')
+        student_file = default_storage.open(path)
+        app_iter = iter(partial(student_file.read, BLOCK_SIZE), '')
         return Response(
             app_iter=app_iter,
             content_type=mimetype,
             content_disposition="attachment; filename=" + filename)
 
+    @XBlock.handler
+    def download_submissions(self, request, suffix=''):
+        require(self.is_course_staff())
+        student_ids = json.loads(request.body).get('student_ids', [])
+        files = self.get_files(self.get_submissions(student_ids))
+        return self.download_zip(files)
+
+    @XBlock.handler
+    def download_all_submissions(self, request, suffix=''):
+        require(self.is_course_staff())
+        files = self.get_files(self.get_submissions())
+        return self.download_zip(files)
+
+    def get_submissions(self, student_ids=None):
+        students = SubmissionsStudent.objects.filter(course_id=self.course_id, item_id=self.block_id)
+        submissions = []
+        for student in students:
+            if not student_ids or student.student_id in student_ids:
```

Comment:
You could do something like `students = students.filter(id__in=student_ids) if student_ids else students` before the loop and drop the condition or `if student_ids: students = students.filter(id__in=student_ids)`

Reply:
Suggestion implemented